lib/state.js

```diff
@@ -468,10 +486,15 @@ StateResolver.prototype.resolveVariable_ = function(name, value) {
   } else if (value.isObject()) {
     data.varTableIndex = this.getVariableIndex_(value);
     var maxProps = this.config_.capture.maxProperties;
-    if (maxProps && maxProps < Object.keys(value.value()).length) {
+    var maxBuffer = this.config_.capture.maxDataSize;
+    var numKeys = Object.keys(value.value()).length;
+
+    if (!isEvaluated && maxProps && maxProps < numKeys) {
       data.status = MESSAGE_TABLE[OBJECT_LIMIT_MESSAGE_INDEX].status;
     }
-
+    else if (isEvaluated && maxBuffer > 0 && maxBuffer < numKeys) {
```

Comment:
I don't know if this test makes a lot of sense. `maxBuffer` is supposed to be the proportional number of bytes in the buffer we send. `numKeys` and `maxBuffer` don't seem like quantities you can compare.

I think what you really want to figure out is how many properties to encode given that the amount of space *left* in the buffer.

Perhaps we should use a different approach here wherein the `maxDataSize` limits are applied only at the top-level in `capture_`. That is, `resolveVariable_` and `resolveMirror_` wouldn't care about `maxDataSize`. WDYT?